app/services/katello/pulp3/repository.rb

```diff
@@ -101,6 +109,14 @@ def repository_versions_api
         PulpcoreClient::RepositoriesVersionsApi.new(core_api_client)
       end
 
+      def uploads_api
+        PulpcoreClient::UploadsApi.new(core_api_client)
+      end
```

Comment:
So today we can only upload 1 type of content per repository type (rpm for yum, file for file, etc...).  This pr: https://github.com/Katello/katello/pull/8099 however changes this assumption and allows the user to pass in what content_type they are wanting to upload.  

So we may want to make this uploads_api exist on the content type service classes themselves.

I also have some concerns that the api may be different across pulp plugins, but we'll just have to see how that goes as we do more :)

Reply:
and i would add, to do this properly, we probably need to pull in his work first, to get some of the changes needed in the repository type manager

Reply:
Pulled in Chris's PR and updated the flow accordingly. This should be ready for review.